numpy/random/mtrand/mtrand.pyx

```diff
@@ -553,299 +553,44 @@ def _shape_from_size(size, d):
-        single value is returned.
-    rngstate : encapsulated pointer to rk_state
-        The specific type depends on the python version. In Python 2 it is
-        a PyCObject, in Python 3 a PyCapsule object.
-
-    Returns
-    -------
-    out : python scalar or ndarray of np.int32
-          `size`-shaped array of random integers from the appropriate
-          distribution, or a single such random int if `size` not provided.
-
-    """
-    cdef npy_uint32 off, rng, buf
-    cdef npy_uint32 *out
-    cdef ndarray array "arrayObject"
-    cdef npy_intp cnt
-    cdef rk_state *state = <rk_state *>NpyCapsule_AsVoidPtr(rngstate)
-
-    rng = <npy_uint32>(high - low)
-    off = <npy_uint32>(<npy_int32>low)
-    if size is None:
-        rk_random_uint32(off, rng, 1, &buf, state)
-        return <npy_int32>buf
-    else:
-        array = <ndarray>np.empty(size, np.int32)
-        cnt = PyArray_SIZE(array)
-        out = <npy_uint32 *>PyArray_DATA(array)
-        with nogil:
-            rk_random_uint32(off, rng, cnt, out, state)
-        return array
-
-
-def _rand_int64(low, high, size, rngstate):
-    """
-    _rand_int64(low, high, size, rngstate)
-
-    See `_rand_int32` for documentation, only the return type changes.
-
-    """
-    cdef npy_uint64 off, rng, buf
-    cdef npy_uint64 *out
-    cdef ndarray array "arrayObject"
-    cdef npy_intp cnt
-    cdef rk_state *state = <rk_state *>NpyCapsule_AsVoidPtr(rngstate)
+# The function bodies ('func') themselves can be found in
+# 'generate_randint_helpers.py'
 
-    rng = <npy_uint64>(high - low)
-    off = <npy_uint64>(<npy_int64>low)
-    if size is None:
-        rk_random_uint64(off, rng, 1, &buf, state)
-        return <npy_int64>buf
-    else:
-        array = <ndarray>np.empty(size, np.int64)
-        cnt = PyArray_SIZE(array)
-        out = <npy_uint64 *>PyArray_DATA(array)
-        with nogil:
-            rk_random_uint64(off, rng, cnt, out, state)
-        return array
-
-def _rand_uint8(low, high, size, rngstate):
-    """
-    _rand_uint8(low, high, size, rngstate)
-
-    See `_rand_int32` for documentation, only the return type changes.
-
-    """
-    cdef npy_uint8 off, rng, buf
-    cdef npy_uint8 *out
-    cdef ndarray array "arrayObject"
-    cdef npy_intp cnt
-    cdef rk_state *state = <rk_state *>NpyCapsule_AsVoidPtr(rngstate)
-
-    rng = <npy_uint8>(high - low)
-    off = <npy_uint8>(low)
-    if size is None:
-        rk_random_uint8(off, rng, 1, &buf, state)
-        return buf
-    else:
-        array = <ndarray>np.empty(size, np.uint8)
-        cnt = PyArray_SIZE(array)
-        out = <npy_uint8 *>PyArray_DATA(array)
-        with nogil:
-            rk_random_uint8(off, rng, cnt, out, state)
-        return array
-
-
-def _rand_uint16(low, high, size, rngstate):
-    """
-    _rand_uint16(low, high, size, rngstate)
-
-    See `_rand_int32` for documentation, only the return type changes.
+def _randint_type(dtype):
```

Comment:
I had issues compiling the code using the dictionary because I place all the functions at the end of the file, so Cython kept complaining that I had all of these undefined variables.